vaadin-date-time-picker-flow-parent/vaadin-date-time-picker-flow/src/main/java/com/vaadin/flow/component/datetimepicker/DateTimePicker.java

```diff
@@ -391,6 +396,118 @@ public String getLabel() {
         return getElement().getProperty("label");
     }
 
+    /**
+     * Set the aria-label of the component to the given text.
+     *
+     * @param ariaLabel
+     *            the aria-label text to set or {@code null} to clear
+     */
+    public void setAriaLabel(String ariaLabel) {
+        getElement().setProperty("accessibleName", ariaLabel);
+    }
+
+    /**
+     * Gets the aria-label of the component.
+     *
+     * @return an optional aria-label of the component if no aria-label has been
+     *         set
+     */
+    public Optional<String> getAriaLabel() {
+        return Optional.ofNullable(getElement().getProperty("accessibleName"));
+    }
+
+    /**
+     * Contains DateTimePicker internalization properties
+     */
+    public static class DateTimePickerI18n implements Serializable {
+        private String dateLabel;
+        private String timeLabel;
+
+        public String getDateLabel() {
+            return dateLabel;
+        }
+
+        public void setDateLabel(String dateLabel) {
+            this.dateLabel = dateLabel;
+        }
+
+        public String getTimeLabel() {
+            return timeLabel;
+        }
+
+        public void setTimeLabel(String timeLabel) {
+            this.timeLabel = timeLabel;
+        }
+    }
+
+    /**
+     * Sets the accessible label to the date picker. The property works in
+     * conjunction with label and aria-label defined on the field. If both
+     * properties are defined, then aria-label takes precedence. Then, the
+     * dateLabel value is concatenated with it.
```

Comment:
I'm not sure I understood the comment all the way. Am I correct that what we try to communicate here is that `aria-label` will take precedence over the date-picker's label when defined?

Reply:
This part refers to the `aria-label` and label set on the `DateTimePicker`, actually.  

This doc indeed needs to be more clear. This is how it works:
- You set only the date-picker label -> date-picker label is `${date-picker label}`
- You set the date-picker and DTP label -> the date-picker `aria-label` is set to `${DTP label} ${date-picker label}`
- You set the date-picker, DTP label, and DTP aria-label -> the date-picker is set to `${DTP aria-label} ${date-picker label}`